src/token-getter.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+const AWS = require('aws-sdk');
+const clui = require('clui');
+
+class TokenGetter {
+  constructor(config, samlAssertion, account, role) {
+    this.spinner = new clui.Spinner('Getting token...');
+    this.sts = new AWS.STS({region: config.region});
+    this.samlAssertion = samlAssertion;
+    this.role = role;
+    this.account = account;
+    this.defaultAccount = config.defaultAccount;
+    this.accountNumber = this.account.accountNumber;
+  }
+
+  async getToken() {
```

Comment:
I'm not a fan of so many things here being `async`. What's the benefit?

Reply:
because JavaScript :( await/async is much better than callbacks and promises.

Reply:
event than the `async` word is used here, this makes the code behave in a more synchronous manner instead of the default async nature of nodejs.

e.g. https://medium.com/tkssharma/writing-neat-asynchronous-node-js-code-with-promises-async-await-fa8d8b0bcd7c

Reply:
The benefit is better readability and less complexity with the same behavior. Eliminates the number of arguments, and callbacks used which increase complexity and are harder to debug and test.

with callbacks:

```
Bafore: lots of nesting and a big function:

sts.assumeRoleWithSAML({	
...
}, function (err, token) {		
    sts.assumeRole({	
        if...
        if...
        ...
    }, function (err, assumedToken) {	
        if...
        ...
    });	
});
```